lib/node_modules/@stdlib/math/base/special/cinvf/lib/main.js

```diff
@@ -0,0 +1,107 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var absf = require( '@stdlib/math/base/special/absf' );
+var maxf = require( '@stdlib/math/base/special/maxf' );
+var f32 = require( '@stdlib/number/float64/base/to-float32' );
+var FLOAT32_BIGGEST = require( '@stdlib/constants/float32/max' );
+var FLOAT32_SMALLEST = require( '@stdlib/constants/float32/smallest-normal' );
+var EPS = require( '@stdlib/constants/float32/eps' );
+var Complex64 = require( '@stdlib/complex/float32/ctor' );
+var realf = require( '@stdlib/complex/float32/real' );
+var imagf = require( '@stdlib/complex/float32/imag' );
+
+
+// VARIABLES //
+
+var LARGE_THRESHOLD = f32( FLOAT32_BIGGEST * f32( 0.5 ) );
+var SMALL_THRESHOLD = f32( FLOAT32_SMALLEST * f32( f32(2.0)/EPS ) );
+var RECIP_EPS_SQR = f32( f32(2.0) / f32(EPS*EPS) );
+
+
+// MAIN //
+
+/**
+* Computes the inverse of a single-precision complex floating-point number.
+*
+* ## References
+*
+* -   Baudin, Michael, and Robert L. Smith. 2012. "A Robust Complex Division in Scilab." _arXiv_ abs/1210.4539 \[cs.MS\] (October): 1–25. <https://arxiv.org/abs/1210.4539>.
+*
+* @param {Complex64} z - complex number
+* @returns {Complex64} result
+*
+* @example
+* var Complex64 = require( '@stdlib/complex/float32/ctor' );
+* var realf = require( '@stdlib/complex/float32/real' );
+* var imagf = require( '@stdlib/complex/float32/imag' );
+*
+* var v = cinvf( new Complex64( 2.0, 4.0 ) );
+* // returns <Complex64>
+*
+* var re = realf( v );
+* // returns ~0.1
+*
+* var im = imagf( v );
+* // returns ~-0.2
+*/
+function cinvf( z ) {
+	var ab;
+	var re;
+	var im;
+	var s;
+	var r;
+	var t;
+
+	re = realf( z );
+	im = imagf( z );
+	ab = maxf( absf(re), absf(im) );
+	s = f32( 1.0 );
```

Comment:
@anandkaranubc Rather than repeatedly do `f32(0.5)` and `f32(1.0)`, can you move those to the parent scope as

```
var ONE = f32( 1.0 );
var HALF = f32( 0.5 );
```